include/dlaf/eigensolver/tridiag_solver/impl.h

```diff
@@ -76,43 +72,27 @@ std::vector<pika::shared_future<T>> cuppensDecomposition(Matrix<T, D>& mat_trd)
   offdiag_vals.reserve(to_sizet(i_end));
 
   for (SizeType i_split = 0; i_split < i_end; ++i_split) {
-    auto sender =
-        ex::when_all(mat_trd(LocalTileIndex(i_split, 0)), mat_trd(LocalTileIndex(i_split + 1, 0)));
-
-    // Cuppen's tridiagonal decomposition
-    offdiag_vals.push_back(
-        di::transform(di::Policy<DefaultBackend<D>::value>(), cuppensDecomp_o, std::move(sender)) |
-        ex::make_future());
+    offdiag_vals.push_back(cuppensDecompAsync<T, D>(mat_trd(LocalTileIndex(i_split, 0)),
+                                                    mat_trd(LocalTileIndex(i_split + 1, 0))));
```

Comment:
I think you can use `readwrite_sender` for both occurrences of the access to `mat_trd`.

Reply:
https://github.com/eth-cscs/DLA-Future/pull/647/commits/f4af88fb772005357257359da21ee7ee4aeeca72